app/src/main/java/ch/hikemate/app/utils/RouteUtils.kt

```diff
@@ -40,29 +40,27 @@ object RouteUtils {
    * Helper function to compute the total elevation gain based on a list of waypoints.
    *
    * @param ways A list of `LatLong` objects representing the waypoints of the hike.
-   * @param hikeId the hikes id, needed for the elevation API request
+   * @param elevationRepository The elevation service to use for computing elevation gain.
    * @return The total elevation gain in meters as a `Double`.
    */
-  fun getElevationGain(
-      ways: List<LatLong>,
-      hikeId: String,
-      elevationService: ElevationServiceRepository
-  ): Double = runBlocking {
+  fun getElevationGain(ways: List<LatLong>, elevationRepository: ElevationRepository): Double =
+      runBlocking {
 
-    // Since elevationService.getElevation is asynchronous, we use a CompletableDeferred to wait for
-    // the result
-    val deferredResult = CompletableDeferred<List<Double>>()
+        // Since elevationRepository.getElevation is asynchronous, we use a CompletableDeferred to
+        // wait
+        // for
+        // the result
```

Comment:
Nitpick: put this on a single line

```suggestion
        // wait for the result
```

Reply:
Good catch!